server/controllers/payroll/reports/payslipGenerator.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * @overview reports/payroll/multipayroll
+ *
+ * @description
+ * This file contains code to create a PDF report of all MultiPayroll registrations,
+ * matching query conditions passed from the multi Payroll UI grid.
+ *
+ * @requires lodash
+ * @requires ReportManager
+ */
+
+const _ = require('lodash');
+
+const ReportManager = require('../../../lib/ReportManager');
+const db = require('../../../lib/db');
+
+const TEMPLATE = './server/controllers/payroll/reports/payslipGenerator.handlebars';
+const PayrollConfig = require('../configuration');
+
+/**
+ * @method build
+ *
+ * @description
+ * This method builds the Payslips for employees payed to be shipped back to
+ * the client.
+ *
+ * GET /reports/payroll/employees
+ */
+function build(req, res, next) {
+  const options = _.clone(req.query);
+  const typeVar = Array.isArray(options.employees);
+
+  options.employees = !typeVar ? [JSON.parse(options.employees)] : options.employees;
+
+  // options.employees = [JSON.parse(options.employees)];
+
+  _.extend(options, {
+    filename : 'FORM.LABELS.PAYSLIP',
+    csvKey : 'payslipGenerator',
+    orientation : 'portrait',
+    footerFontSize : '7',
+  });
+
+  let report;
+  const getPaiementRubrics = [];
+  const getHolidays = [];
+  const getOffDays = [];
+  const employeeData = [];
+
+  const data = {};
+
+  data.enterprise = req.session.enterprise;
+  data.user = req.session.user;
+  data.lang = options.lang;
+
+  // set up the report with report manager
+  try {
+    report = new ReportManager(TEMPLATE, req.session, options);
+    delete options.orientation;
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  options.employees.forEach(emp => {
+    const employee = typeVar ? JSON.parse(emp) : emp;
```

Comment:
If the options.employees  is an Array (When selecting multiple employees)  
    each element of the array must be converted to Json because they are string type